### DISCUSSION TITLE
Unable To Establish Tunnel

### DISCUSSION THREAD

**Discussion by blueaxions** [strongswan/strongswan#1608] [General]
Hello,

I'm running Strongswan on CentOS 8 in FIPS mode but it fails to establish a tunnel. In the logs I see the following error messages. From the articles I found, they all say that the NO PROPOSAL error is due to a configuration mismatch between both sides but for the life of me I can't see any mismatch in my configuration. My test environment is just 2 VMs on the same subnet trying to establish a VPN tunnel, this is only a test environment in my lab, so the network configuration is pretty basic.

The error I see is this:

Mar 17 14:45:46 vpn02 charon-systemd[1674]: received packet: from 10.1.2.4[500] to 10.1.2.3[500] (204 bytes)
Mar 17 14:45:46 vpn02 charon-systemd[1674]: parsed ID_PROT request 0 [ SA V V V V V ]
Mar 17 14:45:46 vpn02 charon-systemd[1674]: no IKE config found for 10.1.2.3...10.1.2.4, sending NO_PROPOSAL_CHOSEN
Mar 17 14:45:46 vpn02 charon-systemd[1674]: generating INFORMATIONAL_V1 request 1580616465 [ N(NO_PROP) ]
Mar 17 14:45:46 vpn02 charon-systemd[1674]: sending packet: from 10.1.2.3[500] to 10.1.2.4[500] (40 bytes)

===================================================================

**My left-right configuration is as follows:**

# ipsec.conf - strongSwan IPsec configuration file

# basic configuration

config setup
        # strictcrlpolicy=yes
        # uniqueids = no
        charondebug="cfg 2, dmn 2, ike 2, net 2"

conn vpnserver
    auto=start
    left=10.1.2.3
    leftid=10.1.2.3
    leftsubnet=0.0.0.0/0
    leftmodecfgserver=yes
    right=10.1.2.4
    rightsourceip=10.10.10.0/22
    rekey=no
    reauth=no
    fragmentation=yes
    authby=secret
    keyexchange=ikev1
    ike=aes256gcm16-aes192gcm16-aes128gcm16-prfsha256-ecp521-ecp256-modp4096-modp2048,aes256-sha256-ecp521-ecp256-modp4096-modp2048!
    esp=aes256gcm16-aes192gcm16-aes128gcm16-ecp521-ecp256-modp4096-modp2048,aes256-sha256-sha1-ecp521-ecp256-modp4096-modp2048, aes256-sha256-sha1!

=========================================================

# ipsec.conf - strongSwan IPsec configuration file

# basic configuration

config setup
        # strictcrlpolicy=yes
        # uniqueids = no
        charondebug="cfg 2, dmn 2, ike 2, net 2"

conn vpnclient
    left=%defaultroute
    leftmodecfgclient=yes
    leftsourceip=%config
    right=10.1.2.3
    rightid=10.1.2.3
    rightsubnet=0.0.0.0/0
    rightmodecfgserver=yes
    narrowing=yes
    rekey=no
    reauth=no
    fragmentation=yes
    authby=secret
    auto=start
    keyexchange=ikev1
    ike=aes256gcm16-aes192gcm16-aes128gcm16-prfsha256-ecp521-ecp256-modp4096-modp2048,aes256-sha256-ecp521-ecp256-modp4096-modp2048!
    esp=aes256gcm16-aes192gcm16-aes128gcm16-ecp521-ecp256-modp4096-modp2048,aes256-sha256-sha1-ecp521-ecp256-modp4096-modp2048, aes256-sha256-sha1!

Any help is greatly appreciated!

-SR

**Comment by tobiasbrunner** [comment]
Please **never** use IKEv1 between two strongSwan hosts. Note that the proposals you configured are technically not valid for IKEv1 as it does not support multiple algorithms per transform type and AES-GCM is not supported for IKEv1 (it can be negotiated for ESP with IKEv1, though). Anyway, check the logs (see #196 for recommended log levels) and make sure the configs are actually loaded because the error you see (`no IKE config found...`)  indicates a more fundamental issue, not the algorithms.

**Comment by blueaxions** [comment]
Thank you for taking the time to respond. This is a temporary test environment that will be torn down shortly. 

I've already changed my configuration from ipsec.conf to swanctl.conf and the Responder and Initiator(s) are working as expected.

You and your team are doing an amazing job supporting Strongswan! We're trying to replace a crappy VPN product with Strongswan and so far, in our experience, Strongswan is a much superior product than what we have currently.

Thanks again for all the hard work in the VPN community and the copious amount of documentation on your wiki page is amazing as well! You guys rock!

-SR

**Comment by blueaxions** [comment]
Thanks.

-SR
